src/Mod/Draft/draftfunctions/move.py

```diff
@@ -215,4 +215,53 @@ def copy_moved_edge(object, edge_index, vector):
         vertex2 = object.Placement.multVec(object.Points[edge_index+1]).add(vector)
     return make_line.make_line(vertex1, vertex2)
 
-## @}
+
+def moveSubElements(obj, sub_objects_names, vector):
+    """moveSubElements(obj, sub_objects_names, vector)
+    
+    Move the given object sub_objects according to a vector or crates an new one
+    if the object is not a Part::Feature.
+
+    Parameters
+    ----------
+    obj : the given object
+
+    sub_objects_names : list of subelement names
+        A list of subelement names to identify the subelements to move.
+
+    vector : Base.Vector
+        Delta Vector to move subobjects from the original position. 
+
+    Return
+    ----------
+    shape : Part.Shape
+        Return the new Shape or None.
+    """
+
+    import Part
+
+    shape = obj.Shape
+    new_shape = None
+
+    replacements = []
+    for sub_objects_name in sub_objects_names:
+        sub_object = obj.Shape.getElement(sub_objects_name)
+        tuple = (zip(sub_object.Vertexes, sub_object.translated(vector).Vertexes))
+        replacements.extend(tuple)
```

Comment:
This loop might cause vertices to be added to 'replacements' multiple times. Not sure what happens then.

Maybe first create a list (set?) of unique vertices. Then create the translated vertices in a final step only and replace ...

Reply:
Yes, you are right. I don't know if we have to add this extra logic or just let the vertex move twice if the user selects both the edge and its subvertexes... It would be annoying but consistent indeed...

Reply:
I didn't even think about the user selecting an edge and a subvertex. I was just thinking about the user selecting two edges sharing a vertex. And the vertex probably wont move twice, because no such vertex is created. But the replace operation on the vertex the second time might run into a missing vertex or something.

Reply:
Uh, I didn't think about that. You are right. I'll have a look at it!